### DISCUSSION TITLE
Handling keypresses

### DISCUSSION THREAD

**Discussion by Napam** [pyedifice/pyedifice#200] [Q&A] [answered]
Hello. I have an app with multiple "views":

```python
@component
def AutoInspector(_):
    initial_history = initialize()
    history, set_history = use_state(initial_history)
    Shared.navigation = Navigation(history, set_history)

    # Very important to have a single "constant" root element or the rendering will be very funny
    with HBoxView():
        match Shared.navigation.current().screen:
            case Screen.Home:
                HomeScreen()
            case Screen.NewProject:
                NewProjectScreen()
            case Screen.Project:
                ProjectScreen()
```
The set_history is shared globally to imitate the browsers history API.

Anyways, when I am in the ProjectScreen (which is just a component), I want to do global actions based on keyboard events. What is the most conventional way to do that using Pyedifice?

**Comment by Napam** [comment, marked as answer]
Never mind, I didn't see that you generally can set a on_key_down, on_mouse_move etc etc on things that inherit edifice.QtWidgetElement.

https://pyedifice.github.io/stubs/edifice.QtWidgetElement.html#edifice.QtWidgetElement

**Comment by jamesdbrock** [comment]
Hi @Napam , I wrote up a little example of how to do global keyboard event handling here.

https://github.com/pyedifice/pyedifice/issues/111

Maybe I should include that code in Edifice? I haven't decided.

**Reply by Napam** [reply]
Yes, I think it would be nice to have that in the docs ☺️. Even though the information is strictly speaking present in the docs, it is not that searchable. Right now when you search for "keyboard", "keypress" and whatnot, in the docs you don't really see anything directly addressing how to handle keypresses, hence this discussion I made.

Like, maybe mention that generally you can specify a on_key_down, on_key_up etc on various components, and then show (or at least refer) to the example you made?
